In [6]:
import pandas as pd
from scipy import stats
import numpy as np

# Datos: Nombre, Altura y Sexo
data = {'Nombre': ['Fer', 'Ale', 'Alex Muñoz', 'Gael', 'Erick', 'Angie', 'Karla', 'Angel', 'Derik', 'Jose Jacal', 
                   'Leo Dorantes', 'Lalo', 'Antonio', 'Luis Hurtado', 'Cristian', 'Juve'],
        'Altura': [180, 170, 170, 168, 165, 168, 170, 175, 174, 175, 175, 172, 170, 177, 170, 176],
        'Sexo': ['M', 'M', 'M', 'M', 'M', 'F', 'F', 'M', 'M', 'M', 'M', 'M', 'M', 'M', 'M', 'M']}

df = pd.DataFrame(data)

# Separar datos por género
mujeres = df[df['Sexo'] == 'F']['Altura']
hombres = df[df['Sexo'] == 'M']['Altura']

In [12]:
# Hipótesis 1: Estatura media de las mujeres comparada con 175 cm
media_mujeres = mujeres.mean()
t_stat, p_value = stats.ttest_1samp(mujeres, 175)
print(f"Hipótesis 1 - Estatura media de las mujeres:")
print(f"Media observada: {media_mujeres}")
print(f"t-statistic: {t_stat}, p-value: {p_value}\n")

Hipótesis 1 - Estatura media de las mujeres:
Media observada: 169.0
t-statistic: -6.0, p-value: 0.10513691342250685



In [11]:
# Hipótesis 2: Desviación estándar de la estatura de las mujeres
desviacion_estandar_mujeres = mujeres.std()
n_mujeres = len(mujeres)
chi2_stat = (n_mujeres - 1) * (desviacion_estandar_mujeres**2) / (10**2)
p_value_chi2 = 1 - stats.chi2.cdf(chi2_stat, n_mujeres - 1)
print(f"Hipótesis 2 - Desviación estándar de las mujeres:")
print(f"Desviación estándar observada: {desviacion_estandar_mujeres}")
print(f"Chi2-statistic: {chi2_stat},\n p-value: {p_value_chi2}\n")

Hipótesis 2 - Desviación estándar de las mujeres:
Desviación estándar observada: 1.4142135623730951
Chi2-statistic: 0.020000000000000004,
 p-value: 0.887537083981715



In [10]:
# Hipótesis 3: Diferencia en la estatura promedio entre hombres y mujeres
t_stat_gender, p_value_gender = stats.ttest_ind(hombres, mujeres)
print(f"Hipótesis 3 - Comparación entre estatura promedio de hombres y mujeres:")
print(f"t-statistic: {t_stat_gender},\n p-value: {p_value_gender}")

Hipótesis 3 - Comparación entre estatura promedio de hombres y mujeres:
t-statistic: 1.2348580371278692,
 p-value: 0.23720767897804637


### Conclusiones

#### Hipótesis 1: Estatura media de las mujeres
- **Media observada:** 169.0 cm
- **t-statistic:** -6.0
- **p-value:** 0.1051

El análisis indica que la media observada de 169.0 cm para la estatura de las mujeres es considerablemente menor que el valor de referencia de 175 cm. Sin embargo, el p-value de 0.1051 es mayor que el nivel de significancia comúnmente utilizado de 0.05, lo que sugiere que no hay suficiente evidencia estadística para rechazar la hipótesis nula. Por lo tanto, no se puede concluir que la estatura media de las mujeres sea significativamente diferente de 175 cm.

#### Hipótesis 2: Desviación estándar de las mujeres
- **Desviación estándar observada:** 1.41 cm
- **Chi2-statistic:** 0.02
- **p-value:** 0.8875

La desviación estándar observada de 1.41 cm es significativamente menor que el valor de referencia de 10 cm. El p-value de 0.8875 también es mucho mayor que 0.05, indicando que no hay evidencia suficiente para rechazar la hipótesis nula de que la desviación estándar es igual o menor que 10 cm. Esto sugiere que la variabilidad en las estaturas de las mujeres es baja y no es estadísticamente significativa.

#### Hipótesis 3: Comparación entre estatura promedio de hombres y mujeres
- **t-statistic:** 1.23
- **p-value:** 0.2372

El p-value de 0.2372 indica que no hay suficiente evidencia para rechazar la hipótesis nula que establece que no hay diferencia significativa entre la estatura promedio de hombres y mujeres. Por lo tanto, no se puede concluir que exista una diferencia en las alturas promedio entre ambos géneros en la muestra analizada.
